### ISSUE TITLE
Would adding an extra RNA barcode to dRNA-Seq require a new HMM model for polya estimates? [question]

### ISSUE THREAD

**Issue by opplatek** [jts/nanopolish#882]
Hi, 
We have direct RNA-Seq where we added an extra RNA barcode sequence after the poly(A) tail. It is 20 nucleotides long with a composition of A 0.15, T 0.25, G 0.3, C 0.3. We would like to use the `nanopolish polya` module but we are not sure whether this doesn't break the assumptions of the HMM model resulting in incorrect poly(A) length estimates. 

I went through the publication note and it says:

> Prior to running the Viterbi algorithm, we apply a global linear rescaling on all samples of the squiggle to remove per-read variations from the base model.

Would this in combination with high weights on the self-loop rather than transition to the next state _remove_ the _effect_ of the added barcode? 

If not, is it possible to retrain the HMM model to include this extra sequence (and how)?

Thank you.

**Comment by jts** [comment]
Hi @opplatek,

This is tricky. The segmentation algorithm is not looking for a precise sequence of bases immediately before the poly-A tail but signals that fall into a fairly broad distribution that are then followed by a fairly uniform segment (the poly-A itself). So it is possible that the signal is segmented correctly, but it is hard to say. 

@paultsw has written a very nice guide to visualizing (and re-training) the HMM here: https://github.com/jts/nanopolish/blob/master/scripts/polya_training/workflow.md. I suggest you follow this guide on some example barcoded reads, then use `hmmplot.py` to see whether the segmentations look good or not.

I hope that helps,
Jared

**Comment by opplatek** [comment]
Thanks, @jts! 

The guide is awesome. I was able to visualize the indexed reads with `hmmplot.py` and it seems the model is able to handle the index without any problems. I have also checked the calls with NOREGION in the qc tag and it looks like the poly(A) detection failed successfully in those. It seems there is no need for retraining the model after all.

Just as an example, here is of one of the indexed reads visualized with the `hmmplot.py` script (bottom) compared to a _regular_ non-indexed read (top).
![_Regular_ non-indexed read segment visualization](https://user-images.githubusercontent.com/17702829/109046076-581ce400-76a2-11eb-91a5-8e0f4e1a7762.png)

![Indexed read segment visualization](https://user-images.githubusercontent.com/17702829/109046123-666b0000-76a2-11eb-8177-88c4196a8310.png)
